c++/src/Statistics.hh

```diff
@@ -41,49 +41,181 @@ namespace orc {
   };
 
 /**
+ * Internal Statistics Implementation
+ */
+
+  template <typename T>
+  class InternalStatisticsImpl {
+  private:
+    bool hasNull_;
+    bool hasMinimum_;
+    bool hasMaximum_;
+    bool hasSum_;
+    bool hasTotalLength_;
+    uint64_t totalLength_;
+    uint64_t valueCount_;
+    T minimum_;
+    T maximum_;
+    T sum_;
+  public:
+    InternalStatisticsImpl() {
+      hasNull_ = false;
+      hasMinimum_ = false;
+      hasMaximum_ = false;
+      hasSum_ = false;
+      hasTotalLength_ = false;
+      totalLength_ = -1;
```

Comment:
0 is a valid value. I think it is safe if we initialize to an invalid value `-1`.